tests/pytests/unit/modules/test_tls.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+import pytest
+import salt.modules.tls as tls
+from tests.support.mock import MagicMock, patch
+
+_TLS_TEST_DATA = {
+    "create_ca": {
+        "bits": 2048,
+        "CN": "localhost",
+        "C": "US",
+        "ST": "Utah",
+        "L": "Salt Lake City",
+        "O": "SaltStack",
+        "OU": "Test Unit",
+        "emailAddress": "xyz@pdq.net",
+        "digest": "sha256",
+        "replace": False,
+    },
+}
+
+
+@pytest.fixture(autouse=True)
+def setup_loader(request):
+    setup_loader_modules = {tls: {}}
+    with pytest.helpers.loader_mock(request, setup_loader_modules) as loader_mock:
+        yield loader_mock
+
+
+def test_create_ca_permissions_on_cert_and_key(tmpdir):
+    ca_name = "test_ca"
+    ca_path = tmpdir
+    certp = "{0}/{1}/{1}_ca_cert.crt".format(ca_path, ca_name)
```

Comment:
Since you're using the `tmpdir` fixture:
```python
certp = tmpdir.join(ca_name).join("{}_ca_cert.crt".format(ca_name)).strpath
certk = tmpdir.join(ca_name).join("{}_ca_cert.key".format(ca_name)).strpath
```

Reply:
Applied!